buffer/src/main/java/io/netty/buffer/PooledUnsafeHeapByteBuf.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+import io.netty.util.Recycler;
+import io.netty.util.Recycler.Handle;
+import io.netty.util.internal.PlatformDependent;
+
+final class PooledUnsafeHeapByteBuf extends PooledHeapByteBuf {
+
+    private static final Recycler<PooledUnsafeHeapByteBuf> RECYCLER = new Recycler<PooledUnsafeHeapByteBuf>() {
+        @Override
+        protected PooledUnsafeHeapByteBuf newObject(Handle handle) {
+            return new PooledUnsafeHeapByteBuf(handle, 0);
+        }
+    };
+
+    static PooledUnsafeHeapByteBuf newUnsafeInstance(int maxCapacity) {
+        PooledUnsafeHeapByteBuf buf = RECYCLER.get();
+        buf.reuse(maxCapacity);
+        return buf;
+    }
+
+    private PooledUnsafeHeapByteBuf(Handle recyclerHandle, int maxCapacity) {
+        super(recyclerHandle, maxCapacity);
+    }
+
+    @Override
+    protected byte _getByte(int index) {
+        return UnsafeByteBufUtil.getByte(memory, idx(index));
+    }
+
+    @Override
+    protected short _getShort(int index) {
+        return UnsafeByteBufUtil.getShort(memory, idx(index));
+    }
+
+    @Override
+    protected int _getUnsignedMedium(int index) {
+        return UnsafeByteBufUtil.getUnsignedMedium(memory, idx(index));
+    }
+
+    @Override
+    protected int _getInt(int index) {
+        return UnsafeByteBufUtil.getInt(memory, idx(index));
+    }
+
+    @Override
+    protected long _getLong(int index) {
+        return UnsafeByteBufUtil.getLong(memory, idx(index));
+    }
+
+    @Override
+    protected void _setByte(int index, int value) {
+        UnsafeByteBufUtil.setByte(memory, idx(index), value);
+    }
+
+    @Override
+    protected void _setShort(int index, int value) {
+        UnsafeByteBufUtil.setShort(memory, idx(index), value);
+    }
+
+    @Override
+    protected void _setMedium(int index, int value) {
+        UnsafeByteBufUtil.setMedium(memory, idx(index), value);
+    }
+
+    @Override
+    protected void _setInt(int index, int value) {
+        UnsafeByteBufUtil.setInt(memory, idx(index), value);
+    }
+
+    @Override
+    protected void _setLong(int index, long value) {
+        UnsafeByteBufUtil.setLong(memory, idx(index), value);
+    }
+
+    @Override
+    protected Recycler<?> recycler() {
+        return RECYCLER;
+    }
+
+    @Override
+    @Deprecated
+    protected SwappedByteBuf newSwappedByteBuf() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)